main.js

```diff
@@ -114,6 +114,20 @@ updateNotifier({pkg: cli.pkg}).notify();
 
 const {input, flags: opts} = cli;
 
+// Make data types for `opts.space` match those of the API
+if (typeof opts.space === 'string') {
+	if (/^\d$/.test(opts.space)) {
```

Comment:
That's actually what I wrote initially and @sindresorhus asked me to remove the `+`, saying that the number would never be larger than 8 (see his second comment on this PR). What's the consensus on this?

Reply:
I didn't see this comment. The problem if we drop the `+` is that `xo --space 10` will be considered as trying to lint the file `10`. I'm not sure this is what we want. @sindresorhus did you considered that case?

Reply:
In addition if we really want to limit that value to `0-8` we should explicitly test it and report the error if it's not valid similarly to https://github.com/xojs/xo/blob/master/main.js#L136

Reply:
Then I would favor making anything more than 8 an error. It makes no sense to do that and is most likely an error.

Reply:
Do we want to throw that error here, though, or should that be in options-manager.js so that the limit applies to both the CLI and the API?

Reply:
Should be in the API.

Reply:
Just to confirm, I'm hearing that we should checking CLI input using `/^\d+$/` (with the `+`) so that we don't treat higher numbers as filenames, then enforce the 8-space limit from the API.

Do you want that limit included in this PR, or should that be separate, since its a different scope?